src/torchcodec/_core/SingleStreamDecoder.h

```diff
@@ -357,6 +356,8 @@ class SingleStreamDecoder {
     // Used to know whether a new FilterGraphContext or UniqueSwsContext should
     // be created before decoding a new frame.
     DecodedFrameContext prevFrameContext;
+
+    std::unique_ptr<DeviceInterface> deviceInterface;
```

Comment:
Let's move this to be a member of `SingleStreamDecoder` based on my previous comment about this class only supporting single stream decoding.

Reply:
done